src/runtime/Types/ClassDerived.cs

```diff
@@ -678,6 +710,127 @@ private static ModuleBuilder GetModuleBuilder(string assemblyName, string module
 
             return moduleBuilder;
         }
+
+        /// <summary>
+        /// Builds a C# attribute based on a python tuple definition (Attibute Type, *attributeArgs, ** attributeKwArgs)
+        /// </summary>
+        static CustomAttributeBuilder GetAttributeBuilder(PyObject attr)
+        {
+            // this is a bit complicated because we want to support both unnamed and named arguments
+            // in C# there is a discrepancy between property setters and named argument wrt attributes
+            // in python there is no difference. But luckily there is rarely a conflict, so we can treat
+            // named arguments and named properties or fields the same way.
+            var tp = new PyTuple(attr);
+            Type attribute = tp[0].As<Type>();
+            if (typeof(Attribute).IsAssignableFrom(attribute) == false)
+            {
+                throw new Exception($"This type cannot be used as an attribute type: {attribute}");
```

Comment:
The exception type must be specific.

Also, please add a test for this scenario to ensure it does not lead to a fatal error. You might need to raise a Python exception instead.